src/synapse.ts

```diff
@@ -207,13 +210,55 @@ export class Synapse {
   }
 
   /**
-   * Gets the signer instance
+   * Gets the signer instance, possibly a session key
    * @returns The ethers signer
    */
   getSigner(): ethers.Signer {
+    if (this._session == null) {
+      return this._signer
+    } else {
+      return this._session.getSigner()
+    }
+  }
+
+  /**
+   * Gets the client signer instance
+   * @returns the ethers signer
+   */
+  getClient(): ethers.Signer {
     return this._signer
   }
 
+  /**
+   * Sets the signer as the session key for storage actions
+   * @param sessionKeySigner The session key signer
+   * @returns The SessionKey object for this signer
+   * @example
+   * ```typescript
+   * const sessionKey = synapse.setSession(privateKey)
+   *
+   * // check for previous login
+   * await sessionKey.fetchExpiries()
+   * const HOUR_MILLIS = 1000 * 60 * 60
+   * if (sessionKey.expiries[ADD_PIECES_TYPEHASH] * 1000 < Date.now() + HOUR_MILLIS) {
+   *   const DAY_MILLIS = 24 * HOUR_MILLIS
+   *   const loginTx = await sessionKey.login(BigInt(Date.now()) / BigInt(1000 + 30 * DAY_MILLIS), PDP_PERMISSIONS)
+   *   const loginReceipt = await loginTx.wait()
+   * }
+   *
+   * const context = await synapse.storage.createContext()
+   * ```
+   */
+  setSession(sessionKeySigner: ethers.Signer): SessionKey {
+    this._session = new SessionKey(
+      this._provider,
+      this._warmStorageService.getSessionKeyRegistry(),
+      sessionKeySigner,
+      this._signer
+    )
+    return this._session
+  }
+
```

Comment:
this should get a SessionKey instance not a ethers.Signer so the developer can manage the SessionKey creation and validation from the outside